nixos/lib/systemd-lib.nix

```diff
@@ -169,6 +172,30 @@ in rec {
     optional (attr ? ${name} && !isInt attr.${name})
       "Systemd ${group} field `${name}' is not an integer";
 
+  assertNftSet = name: group: attr:
+    let
+      defs = map (splitString ":") (filter (x: x != "") (splitString " " attr.${name}));
```

Comment:
`This option expects a whitespace separated list of NFT set definitions. Each definition consists of a colon-separated tuple of source type (one of "address", "prefix" or "ifindex"), NFT address
           family (one of "arp", "bridge", "inet", "ip", "ip6", or "netdev"), table name and set name.`

I haven't tested everything yet, but this seemed to imply that you can put multiple definitions on one line e.g.
```
NFTSet=address:inet:mytable:myset address:inet:mytable:myset2
```
If it's not the case I'll remove it.

I wasn't sure about checks on the length. With the nftables tools I seem to be able to create very long set names despite apparent checks on `IPSET_MAXNAMELEN` in the kernel. It appears that you found limits in systemd though.

Reply:
I think irrespective of the validity of a single whitespace separated list, it may be better to force the list style definition as in
```nix
NFTSet = [
  "address:inet:mytable:myset
  "address:inet:mytable:myset2'
];
```
I was trying to go overboard with accepting whatever systemd does, but it's probably better to start off more restrictive.